out_azure_logs_ingestion: add support for Managed Identities

stefanoboriero opened the pull request:
This change updates the documentation to document support for Managed Identities authentication. It tries to align with the documentation style and content for the similar feature for the out_azure_kusto plugin. The feature is implemented on PR https://github.com/fluent/fluent-bit/pull/10867

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Documentation**
  * Added an "Authentication methods" section covering Service Principal and Managed Identity workflows.
  * Inserted a dedicated Service Principal subsection and clarified configuration defaults (compress now defaults to true).
  * Added complete example configurations for user-assigned and system-assigned managed identities in supported formats.

<!-- review_stack_entry_start -->

[![Review Change Stack](https://storage.googleapis.com/coderabbit_public_assets/review-stack-in-coderabbit-ui.svg)](https://app.coderabbit.ai/change-stack/fluent/fluent-bit-docs/pull/2062?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

<!-- review_stack_entry_end -->
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

stefanoboriero committed:
out_azure_logs_ingestion: add support for Managed Identities

This change updates the documentation to document support for Managed
Identities authentication. It tries to align with the documentation
style and content for the similar feature for the out_azure_kusto
plugin.

Signed-off-by: Stefano Boriero <stefano.boriero@seqera.io>

esmerel reviewed (commented):
Stylistic updates for consistency. @fluent/fluent-bit-maintainers should review for technical accuracy.

esmerel reviewed (commented):
Stylistic updates for consistency. @fluent/fluent-bit-maintainers should review for technical accuracy.

eschabell committed:
Merge branch 'master' into azure-log-ingestion-msi

Signed-off-by: Eric D. Schabell <eric@schabell.org>

eschabell committed:
Update pipeline/outputs/azure_logs_ingestion.md

Co-authored-by: Lynette  Miles <6818907+esmerel@users.noreply.github.com>
Signed-off-by: Eric D. Schabell <eric@schabell.org>

eschabell commented on pipeline/outputs/azure_logs_ingestion.md:
```diff
@@ -21,13 +21,37 @@ To get more details about how to set up these components, refer to the following
 | :------------ | :------------------------- | :------ |
 | `tenant_id`    | The tenant ID of the Azure Active Directory (AAD) application. | _none_ |
-| `client_id`    | The client ID of the AAD application. | _none_ |
+| `client_id`    | _Required for service_principal and managed_identity auth_ - The client ID of the AAD registered application. When using managed identity authentication, set this to 'system' for system-assigned identity or provide the managed identity's client ID. | _none_ |
```
outdated so resolving.

eschabell committed:
Update pipeline/outputs/azure_logs_ingestion.md

Co-authored-by: Lynette  Miles <6818907+esmerel@users.noreply.github.com>
Signed-off-by: Eric D. Schabell <eric@schabell.org>

eschabell committed:
Update pipeline/outputs/azure_logs_ingestion.md

Co-authored-by: Lynette  Miles <6818907+esmerel@users.noreply.github.com>
Signed-off-by: Eric D. Schabell <eric@schabell.org>

eschabell committed:
Update pipeline/outputs/azure_logs_ingestion.md

Co-authored-by: Lynette  Miles <6818907+esmerel@users.noreply.github.com>
Signed-off-by: Eric D. Schabell <eric@schabell.org>

eschabell commented on pipeline/outputs/azure_logs_ingestion.md:
```diff
@@ -21,14 +21,37 @@ To get more details about how to set up these components, refer to the following
+
+### Service principal authentication 
+
+Service principal authentication is the default method. To use it, you mst create an Azure AD application:
```
@stefanoboriero small spelling error here.. think it should be "must" instead of "mst" in this line?

eschabell replied:
Fixing.

eschabell commented on pipeline/outputs/azure_logs_ingestion.md:
```diff
@@ -21,14 +21,37 @@ To get more details about how to set up these components, refer to the following
+
+### Managed identity authentication
+
+When running on Azure services that support managed identities (such as Azure VMs, AKS, or App Service):
```
@stefanoboriero this is a vale issue and can be fixed by using back-ticks around the offending acronym(s).

eschabell replied:
Fixing.

eschabell commented on pipeline/outputs/azure_logs_ingestion.md:
```diff
@@ -21,14 +21,37 @@ To get more details about how to set up these components, refer to the following
+1. [Assign the managed identity appropriate permissions to your Kusto database](https://learn.microsoft.com/en-us/azure/data-explorer/configure-managed-identities-cluster).
+1. Configure Fluent Bit with `auth_type` set to `managed_identity`.
+1. For system-assigned identity, set `client_id` to `system`.
+1. For user-assigned identity, set `client_id` to the managed identity's client ID (GUID).
```
@stefanoboriero another back-ticks need for the acronym here.

eschabell replied:
Fixing.

eschabell reviewed (changes requested):
@stefanoboriero thanks so much for the docs PR, just a few trivial vale issues I tagged that need your attention. Update these and tag me to re-review.

eschabell commented on pipeline/outputs/azure_logs_ingestion.md:
```diff
@@ -21,14 +21,37 @@ To get more details about how to set up these components, refer to the following
+
+### Service principal authentication 
+
+Service principal authentication is the default method. To use it, you mst create an Azure AD application:
```
Fixing.

eschabell replied:
Fixed.

eschabell commented on pipeline/outputs/azure_logs_ingestion.md:
```diff
@@ -21,14 +21,37 @@ To get more details about how to set up these components, refer to the following
+
+- [Register an Application](https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app#register-an-application)
+- [Add a client secret](https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app#add-a-client-secret)
+- [Authorize the app in your database](https://docs.microsoft.com/en-us/azure/data-explorer/kusto/management/access-control/principals-and-identity-providers#azure-ad-tenants)
```
Fixing.

eschabell commented on pipeline/outputs/azure_logs_ingestion.md:
```diff
@@ -134,4 +159,168 @@ pipeline:
+  Name            azure_logs_ingestion
+  Match           sample
+  client_id       system
+  auth_type       managed_identity
```
Fixing.

eschabell commented on pipeline/outputs/azure_logs_ingestion.md:
```diff
@@ -21,14 +21,37 @@ To get more details about how to set up these components, refer to the following
+
+When running on Azure services that support Managed Identities (such as Azure VMs, AKS, or App Service):
+
+1. [Assign the managed identity appropriate permissions to your Kusto database](https://learn.microsoft.com/en-us/azure/data-explorer/configure-managed-identities-cluster)
```
Fixing.

eschabell committed:
docs: pipeline: outputs: azure_logs_ingestion: general doc updates and cleanup

  - Fix typo: mst -> must in service principal section
  - Spell out AKS acronym on first use (Azure Kubernetes Service)
  - Replace GUID acronym with full term "globally unique identifier"
  - Replace Kusto auth link with DCR permissions tutorial link and update link text
  - Convert .conf plugin keys to Title_Case

  Applies to #2062

Signed-off-by: Eric D. Schabell <eric@schabell.org>

eschabell reviewed (approved):
@stefanoboriero just waiting now on code PR merging.
